From 845047c2d43894ac49be3c051a73151dbe277ca1 Mon Sep 17 00:00:00 2001
From: Tonyliu <Bo.Liu@windriver.com>
Date: Mon, 11 Oct 2010 16:11:17 +0800
Subject: [PATCH 5/5] fsl_mpc8308_rdb: update DTS to support DMA controller

The DMA controller on mpc8308 is compatible with it on mpc5121e
instead of Freescale Elo DMA engine. Correct DMA related definitions
in DTS to support DMA.

Signed-off-by: Tonyliu <Bo.Liu@windriver.com>
---
 arch/powerpc/boot/dts/mpc8308_rdb.dts |   40 +++-----------------------------
 1 files changed, 4 insertions(+), 36 deletions(-)

diff --git a/arch/powerpc/boot/dts/mpc8308_rdb.dts b/arch/powerpc/boot/dts/mpc8308_rdb.dts
index f9db6f6..6862004 100644
--- a/arch/powerpc/boot/dts/mpc8308_rdb.dts
+++ b/arch/powerpc/boot/dts/mpc8308_rdb.dts
@@ -153,43 +153,11 @@
 			mode = "cpu";
 		};
 
-		dma@82a8 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			compatible = "fsl,mpc8308-dma", "fsl,elo-dma";
-			reg = <0x82a8 4>;
-			ranges = <0 0x8100 0x1a8>;
+		dma@2c000 {
+			compatible = "fsl,mpc8308-dma", "fsl,mpc5121-dma";
+			reg = <0x2c000 0x2000>;
+			interrupts = <3 0x8 94 0x8>;
 			interrupt-parent = <&ipic>;
-			interrupts = <71 8>;
-			cell-index = <0>;
-			dma-channel@0 {
-				compatible = "fsl,mpc8308-dma-channel", "fsl,elo-dma-channel";
-				reg = <0 0x80>;
-				cell-index = <0>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@80 {
-				compatible = "fsl,mpc8308-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x80 0x80>;
-				cell-index = <1>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@100 {
-				compatible = "fsl,mpc8308-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x100 0x80>;
-				cell-index = <2>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@180 {
-				compatible = "fsl,mpc8308-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x180 0x28>;
-				cell-index = <3>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
 		};
 
 		usb@23000 {
-- 
1.6.5.2

